public/js/components/SourceSearch.js

```diff
@@ -83,17 +91,20 @@ const Search = React.createClass({
       Autocomplete({
         selectItem: result => {
           this.props.selectSource(result.id);
+          this.setState({ previousInput: "" });
           this.props.toggleFileSearch(false);
         },
         handleClose: this.close,
-        items: searchResults(this.props.sources)
+        items: searchResults(this.props.sources),
+        previousInput: this.state.previousInput
```

Comment:
could we call this `searchValue` or `inputValue`? Sometimes with this stuff, i wonder if we need the tense for instance does "previous" make it more clear or is "current" redundant... The answer could well be, previous helps for X reason, i just haven't thought about it much.